shoop/core/pricing/__init__.py

```diff
@@ -155,15 +146,18 @@ def get_pricing_steps(self, context, product_id):
         """
         return [(0, TaxlessPrice(0))]
 
-    def get_prices(self, context, product_ids, quantity=1):
-        return {
-            product_id: self.get_price(context, product_id=product_id, quantity=quantity)
-            for product_id in product_ids
-        }
+    def get_price_infos(self, context, products, quantity=1):
+        """
+        :param products: a list of `Product`s or id's
+        :type products:  list
+        """
+        if not all(isinstance(item, six.integer_types) for item in products):
```

Comment:
I'm not a doctor :P

Anyway, you could do something like this
```python
product_ids = [getattr(x, "pk", x) for x in products]
```

And that would allow even mixing Products and product ids to the same iterable.